platform.txt

```diff
@@ -172,3 +172,11 @@ tools.serial_upload.path.linux64={runtime.hardware.path}/tools/linux64
 tools.serial_upload.upload.params.verbose=-d
 tools.serial_upload.upload.params.quiet=n
 tools.serial_upload.upload.pattern="{path}/{cmd}" {serial.port.file} {upload.altID} {upload.usbID} "{build.path}/{build.project_name}.bin"
+
+# blackmagic upload for generic STM32
+tools.bmp_upload.cmd=arm-none-eabi-gdb
+tools.bmp_upload.path={runtime.tools.arm-none-eabi-gcc.path}/bin/
```

Comment:
I think you should use
`tools.bmp_upload.path={compiler.path}`

which is set on top of platform.txt
`compiler.path={runtime.tools.arm-none-eabi-gcc-6-2017-q2-update.path}/bin/`
This will avoid any issue if compiler version change